Extension/src/LanguageServer/client.ts

```diff
@@ -745,6 +751,43 @@ class DefaultClient implements Client {
         }
     }
 
+    private promptCompileCommands(params: CompileCommandsPaths) : void {
+        if (this.configuration.Configurations[this.configuration.CurrentConfiguration].compileCommands !== undefined) {
+            return;
+        }
+
+        let showCompileCommandsSelection: PersistentState<boolean> = new PersistentState<boolean>("CPP.showPromptCompileCommands", true);
```

Comment:
PersistentState is global.  We might only want to disable it for the current workspace.  You can use `PersistentFolderState` for that.